searchcore/src/vespa/searchcore/proton/server/storeonlyfeedview.cpp

```diff
@@ -602,14 +630,14 @@ StoreOnlyFeedView::adjustMetaStore(const DocumentOperation &op, const DocumentId
                 op.getValidPrevDbdId(_params._subDbId) &&
                 op.getLid() != op.getPrevLid())
             {
-                moveMetaData(_metaStore, docId, op);
+                moveMetaData(_metaStore, docId.getGlobalId(), op);
             } else {
-                putMetaData(_metaStore, docId, op, _params._subDbType == SubDbType::REMOVED);
+                putMetaData(_metaStore, gid, docId, op, _params._subDbType == SubDbType::REMOVED);
```

Comment:
putMetaData should not be called for RemoveOperationWithGid. But this should be implicitly enforced by the operation always having 0 (i.e. invalid) as the new lid.

Reply:
Fixed